build/android/generate_xwalk_shared_library.py

```diff
@@ -0,0 +1,251 @@
+    if src_components[0] != 'drawable' or 'mdpi' not in src_components:
+      continue
+    src_dir = os.path.join(res_root, src_dir_name)
+    if not os.path.isdir(src_dir):
+      continue
+    dst_components = [c for c in src_components if c != 'mdpi']
+    assert dst_components != src_components
+    dst_dir_name = '-'.join(dst_components)
+    dst_dir = os.path.join(res_root, dst_dir_name)
+    if not os.path.isdir(dst_dir):
+      os.makedirs(dst_dir)
+    for src_file_name in os.listdir(src_dir):
+      if not src_file_name.endswith('.png'):
+        continue
+      src_file = os.path.join(src_dir, src_file_name)
+      dst_file = os.path.join(dst_dir, src_file_name)
+      assert not os.path.lexists(dst_file)
+      shutil.move(src_file, dst_file)
+
+
+def ReplaceCrunchedImage(project_source, filename, filepath):
+  """Replace crunched images with source images.
+  """
+  search_dir = [
+      'content/public/android/java/res',
+      'ui/android/java/res'
+  ]
+
+  pathname = os.path.basename(filepath)
+  #replace crunched 9-patch image resources.
+  for search in search_dir:
+    absdir = os.path.join(project_source, search)
+    for dirname, _, files in os.walk(absdir):
+      if filename in files:
+        relativedir = os.path.basename(dirname)
+        if (pathname == 'drawable' and relativedir == 'drawable-mdpi') or \
+            relativedir == pathname:
+          source_file = os.path.abspath(os.path.join(dirname, filename))
+          target_file = os.path.join(filepath, filename)
+          shutil.copyfile(source_file, target_file)
+          return
+
+
+def CopyResources(project_source, out_dir, out_project_dir):
+  print 'Copying resources...'
+  res_dir = os.path.join(out_project_dir, 'res')
+  temp_dir = os.path.join(out_project_dir, 'temp')
+  if os.path.exists(res_dir):
+    shutil.rmtree(res_dir)
+  if os.path.exists(temp_dir):
+    shutil.rmtree(temp_dir)
+
+  # All resources should be in specific folders in res_directory.
+  # Since there might be some resource files with same names from
+  # different folders like ui_java, content_java and others,
+  # it's necessary to rename some files to avoid overridding.
+  res_to_copy = [
+      # zip file list
+      'xwalk_app_strings.zip',
+      'xwalk_core_java.zip'
+  ]
+
+  for res_zip in res_to_copy:
+    zip_file = os.path.join(out_dir, 'res.java', res_zip)
+    zip_name = os.path.splitext(res_zip)[0]
+    if not os.path.isfile(zip_file):
+      raise Exception('Resource zip not found: ' + zip_file)
+    subdir = os.path.join(temp_dir, zip_name)
+    if os.path.isdir(subdir):
+      raise Exception('Resource zip name conflict: ' + zip_name)
+    os.makedirs(subdir)
+    with zipfile.ZipFile(zip_file) as z:
+      z.extractall(path=subdir)
+    CopyDirAndPrefixDuplicates(subdir, res_dir, zip_name)
+    MoveImagesToNonMdpiFolders(res_dir)
+
+  if os.path.isdir(temp_dir):
+    shutil.rmtree(temp_dir)
+
+  #search 9-patch, then replace it with uncrunch image.
```

Comment:
Ditto, the code around here is just copied from generate_xwalk_core_library, I need more investigation to know the purpose